test/e2e/system_connection_test.go

```diff
@@ -313,6 +335,86 @@ qe ssh://root@podman.test:2222/run/podman/podman.sock ~/.ssh/id_rsa false true
 		})
 	})
 
+	Context("with running API service", func() {
+		BeforeEach(func() {
+			SkipIfContainerized("fails for some reason inside containers")
+			SkipIfRemote("connection heuristic requires both podman and podman-remote binaries")
+			SkipIfSystemdNotRunning("cannot test connection heuristic if systemd is not running")
+		})
+
+		It("add tcp:// connection with reverse proxy path", func() {
+			u, err := user.Current()
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Ensure that the remote end uses our built podman
+			if os.Getenv("PODMAN_BINARY") == "" {
+				err = os.Setenv("PODMAN_BINARY", podmanTest.PodmanBinary)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				defer func() {
+					os.Unsetenv("PODMAN_BINARY")
+				}()
+			}
+
+			// Create a reverse proxy to the podman socket using path prefix
+			const pathPrefix = "/reverse/proxy/path/prefix"
+			proxy := http.NewServeMux()
+			proxy.Handle(pathPrefix+"/", &httputil.ReverseProxy{
+				Rewrite: func(pr *httputil.ProxyRequest) {
+					pr.Out.URL.Path = strings.TrimPrefix(pr.Out.URL.Path, pathPrefix)
+					pr.Out.URL.RawPath = strings.TrimPrefix(pr.Out.URL.RawPath, pathPrefix)
+				},
+				Transport: &http.Transport{
+					DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+						podmanSock := fmt.Sprintf("/run/user/%s/podman/podman.sock", u.Uid)
+						return (&net.Dialer{}).DialContext(ctx, "unix", podmanSock)
+					},
+				},
+			})
+			srv := &http.Server{
+				Handler:           proxy,
+				ReadHeaderTimeout: time.Second,
+			}
+
+			// Serve the reverse proxy on a random port
+			lis, err := net.Listen("tcp", "localhost:0")
+			Expect(err).ToNot(HaveOccurred())
+			defer lis.Close()
+
+			defer srv.Close()
+			go func() {
+				defer GinkgoRecover()
+				Expect(srv.Serve(lis)).To(MatchError(http.ErrServerClosed))
+			}()
+
+			connectionURL := "tcp://" + lis.Addr().String() + pathPrefix
+
+			cmd := exec.Command(podmanTest.RemotePodmanBinary,
+				"system", "connection", "add",
+				"--default", "QA", connectionURL,
+			)
+			session, err := Start(cmd, GinkgoWriter, GinkgoWriter)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%q failed to execute", podmanTest.RemotePodmanBinary))
+			Eventually(session, DefaultWaitTimeout).Should(Exit(0))
+			Expect(session.Out.Contents()).Should(BeEmpty())
+			Expect(session.Err.Contents()).Should(BeEmpty())
+
+			cmd = exec.Command(podmanTest.RemotePodmanBinary,
+				"--connection", "QA", "ps")
+			_, err = Start(cmd, GinkgoWriter, GinkgoWriter)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
this isn't testing anything because you just exec the command but never check the output or exit status

Overall on closer look most of this test file seems completely broken, connecting to the host socket is certainly not correct (because that just makes an invalid assumption that the host has a working socket) and /run/user/0/podman/podman.sock is not the correct socket path for root either.

And even when I fix this locally the client still gets 502 so the proxy code doesn't seem to be working either. I don't even see the DialContext function being called so I don't think we hit the handler at all.

Reply:
Thanks for testing my tests and giving detailed feedback. I got a chance to revisit the tests and I think I'm much closer now.

I'll request re-review once I believe the tests are correct and passing.